tests/run/unused-24.scala

```diff
@@ -0,0 +1,23 @@
+object Test {
+
+  def main(args: Array[String]): Unit = {
+    println(fun(new Bar))
+  }
+
+  def fun(unused foo: Foo): foo.X = {
```

Comment:
This tests is unsound because it treats ghost values as stable (#4060) /cc @nicolasstucki